test/unittest/xmlparser/XMLElementParserTests.cpp

```diff
@@ -2123,4 +2293,248 @@ TEST_F(XMLParserTests, getXMLDiscoverySettings_invalidXML)
     EXPECT_EQ(XMLP_ret::XML_ERROR, XMLParserTest::getXMLDiscoverySettings_wrapper(titleElement,settings,ident));
 }
 
+/*
+ * This test checks the negative cases in the xml child element of <SendBuffersAllocationAttributes>
+ * 1. Check an invalid tag of:
+ *      <preallocated_number>
+ *      <dynamic>
+ * 2. Check invalid element
+ */
+TEST_F(XMLParserTests, getXMLSendBuffersAllocationAttributes_invalidXML)
+{
+    uint8_t ident = 1;
+    rtps::SendBuffersAllocationAttributes allocation;
```

Comment:
```suggestion
    SendBuffersAllocationAttributes allocation;
```
Already in the `eprosima::fastrtps::rtps` namespace